stream2hop/tns2hop.py

```diff
@@ -0,0 +1,251 @@
+                str(spectra["source_group"]["id"])
+            ]
+
+        if "instrument" in spectra:
+            spectra["instrument"]["description"] = parameters_list["instrument"][
+                str(spectra["instrument"]["id"])
+            ]
+
+        if "telescope" in spectra:
+            spectra["telescope"]["description"] = parameters_list["telescope"][
+                str(spectra["telescope"]["id"])
+            ]
+
+        if "assoc_groups" in spectra:
+            for assoc_group in spectra["assoc_groups"]:
+                assoc_group["description"] = parameters_list["groups"][
+                    str(assoc_group["id"])
+                ]
+
+    return object
+
+
+def get_new_data(hop_url, config, api_key):
+    """
+      Retrieve new TNS objects
+
+      Args:
+        hop_url: hop URL that will be used to publish data to it
+        config: configurations to access hop URL
+
+      Returns:
+        None
+    """
+
+    url_tns_api = "https://wis-tns.weizmann.ac.il/api/get/"
+
+    #  every 3 hours
+    days_ago = 0.125
+    date = str(datetime.utcnow() - timedelta(days=days_ago))
+    print("Running on: " + date)
+    #  Set search parameter
+    search_obj = {"public_timestamp": date}
+    response = search(url_tns_api, search_obj, api_key)
+
+    if None not in response:
+        print("hop_url = " , hop_url)
+        #  New data is retrieved, open a stream with hop
+        sC = ut.HopConnection(hop_url, config)
+        sC.open()
+
+        #  Read parameters file
+        parameters_list = json.load(open("../parameters.json", "r"))["data"]
+        objects_list = json.loads(response.text)["data"]["reply"]
+
+        for object in objects_list:
+            get_obj = {
+                "objname": object["objname"],
+                "photometry": "1",
+                "spectra": "1",
+                "classification": "1",
+                "discovery": "1",
+                "AT": "1",
+            }
+            response = get_object(url_tns_api, get_obj, api_key)
+
+            if response:
+                object_data = json.loads(response.text)["data"]["reply"]
+                object_data = fix_photometry(object_data, parameters_list)
+                object_data = fix_spectra(object_data, parameters_list)
+                object_data = {"format": "BLOB", "content": object_data}
+                sC.write(json.dumps(object_data, indent=4))
+        sC.close()
+    else:
+        print("Nothing to do\n")
+
+    pass
+
+
+def get_astronotes(hop_url, config, api_key):
+    """
+        Get the latest astronotes in the previous day
+
+        Args:
+            hop_url: hop URL that will be used to publish data to it
+            config: configurations to access hop URL
+
+        Returns:
+            None
+    """
+    date = datetime.today().strftime("%Y-%m-%d")
+    astronotes_url = (
+        "https://wis-tns.weizmann.ac.il/astronotes?&date_start%5Bdate%5D="
+        + date
+        + "&format=csv"
+    )
```

Comment:
Use `urljoin` here as well.

Reply:
I tried to use urljoin here as something like this 
` urljoin("https://wis-tns.weizmann.ac.il/astronotes?&date_start%5Bdate%5D=", "2020-10-01", "&format=csv")` 
but the result was `'https://wis-tns.weizmann.ac.il/2020-10-01`
It concatenates with the slash and not as what the link should be.
Do you have a suggestion for this?

Reply:
Oh I see, I should've looked more closely. `urljoin` is for concatenating two urls, but doesn't do anything with query params. You're doing the param query manipulation and combining it with string concatenation. I'm not sure how to highlight the line after the one highlighted, but you can use `requests` to pass in query params separately like so:

```
astronotes_url = "https://wis-tns.weizmann.ac.il/astronotes"
params = {"date_start[date]": date, "format": "csv"}
response = requests.get(astronotes_url, stream=True)
```

and it should encode the relevant characters properly (`date_start[date]` -> `date_start%5Bdate%5D`).

Reply:
Awesome!!!
Done